doc/source/whatsnew/v1.1.0.rst

```diff
@@ -90,6 +90,8 @@ Backwards incompatible API changes
   now raise a  ``TypeError`` if a not-accepted keyword argument is passed into it.
   Previously a ``UnsupportedFunctionCall`` was raised (``AssertionError`` if ``min_count`` passed into :meth:`~DataFrameGroupby.median``) (:issue:`31485`)
 - :meth:`DataFrame.at` and :meth:`Series.at` will raise a ``TypeError`` instead of a ``ValueError`` if an incompatible key is passed, and ``KeyError`` if a missing key is passed, matching the behavior of ``.loc[]`` (:issue:`31722`)
+- Passing an integer dtype other than ``int64`` to ``np.array(period_index, dtype=...)`` will now raise ``NotImplementedError`` instead of incorrectly using ``int64`` (:issue:`32255`)
```

Comment:
Why NotImplementedError, rathan something like ValueError? Are we the ones raising it, or is NumPy?

Reply:
Though from the tests, it looks like a TypeError is raised (unless I'm misreading).